src/bors/handlers/review.rs

```diff
@@ -122,10 +155,27 @@ pub(super) async fn handle_push_to_pull_request(
     notify_of_pushed_pr(&repo_state, pr_number, pr.head.sha.clone()).await
 }
 
-fn sufficient_approve_permission(repo: Arc<RepositoryState>, author: &GithubUser) -> bool {
-    repo.permissions
+async fn has_review_or_delegated_permission(
+    repo_state: &RepositoryState,
+    author: &GithubUser,
+    pr: &PullRequest,
+    db: &PgDbClient,
+) -> anyhow::Result<bool> {
+    let result = if repo_state
+        .permissions
         .load()
         .has_permission(author.id, PermissionType::Review)
+    {
+        true
+    } else {
+        let pr_model = db
+            .get_or_create_pull_request(repo_state.repository(), pr.number)
+            .await?;
+
+        pr_model.delegated
```

Comment:
We also need to check if the author of the comment is the same as the author of the PR :sweat_smile: By comparing their GitHub IDs. And add a test for this.